packages/relay-compiler/transforms/DeclarativeConnectionMutationTransform.js

```diff
@@ -65,11 +65,18 @@ function visitScalarField(field: ScalarField): ScalarField {
     return field;
   }
   const schema = this.getContext().getSchema();
-  if (!schema.isId(field.type)) {
+
+  if (
+    !schema.isId(field.type) &&
+    (!schema.isList(field.type) ||
+      (schema.isList(field.type) && !schema.isId(field.type.ofType)))
```

Comment:
This results in a Flow error, since I couldn't figure out how to properly refine `field.type` into a list of `ID`s. Any guidance on how to refine that type properly using `schema` is appreciated.